servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -73,26 +81,40 @@ void handleSubscribe(Subscriber subscriber, AsyncContextMap contextMap,
         private static final AtomicIntegerFieldUpdater<AbstractOffloadedSingleValueSubscriber> stateUpdater =
                 newUpdater(AbstractOffloadedSingleValueSubscriber.class, "state");
 
+        private final BooleanSupplier shouldOffload;
         final Executor executor;
         @Nullable
         // Visibility: Task submitted to executor happens-before task execution.
         private Cancellable cancellable;
         @Nullable
         private Object terminal;
         private volatile int state = STATE_INIT;
+        private boolean hasOffloaded;
 
-        AbstractOffloadedSingleValueSubscriber(final Executor executor) {
+        AbstractOffloadedSingleValueSubscriber(final BooleanSupplier shouldOffload, final Executor executor) {
+            this.shouldOffload = shouldOffload;
```

Comment:
`requireNonNull(shouldOffload)`?

Reply:
I put it in the `BooleanSupplier shouldOffload()` method instead.